renderdoc/serialise/rdcfile.h

```diff
@@ -52,11 +53,12 @@ class RDCFile
   // version number of overall file format or chunk organisation. If the contents/meaning/order of
   // chunks have changed this does not need to be bumped, there are version numbers within each
   // API that interprets the stream that can be bumped.
-  static const uint32_t SERIALISE_VERSION = 0x00000100;
+  static const uint32_t SERIALISE_VERSION = 0x00000101;
```

Comment:
I don't want to break container compatibility lightly. I generally don't keep forward compatibility for individual API serialisation etc, but bumping this version means any older build will not even be able to open .rdc files at all to inspect what driver is in use, etc.

Thinking about the alternatives for doing this in a compatible way, I think my preferred solution is to add the new typed thumbnail as a new registered container file section. See the `SectionType` enum for the pre-defined sections that currently exist - e.g. `FrameCapture` for the capture itself, `Notes` and `Bookmarks` for saving persisted UI data.

The closest analog is `ResolveDatabase` which is written automatically at the end of captures that have callstacks embedded, with resolve information for the addresses. If you added a new section type `ExtendedThumbnail` or something like that, you could store the new typed thumbnail in a section in a similar way that's then transparently ignored by older versions of RenderDoc, but can be used by new ones. `CaptureFile::GetThumbnail` can look for the section and use it if present.

As a side-note, this is something I should really have done for 1.0 when refactoring this area, but it doesn't really make sense anymore to have the image resizing and jpeg compression code happening duplicated in each API's backend, especially if you're now making fixes/tweaks in 4 places. It would make more sense just to hand over the raw unresized unencoded (but maybe normalised to RGBA8) screenshot to `CreateRDC` and have any resize/encode happen once inside there. Up to you if you want to do this yourself while refactoring to save the thumbnail as a section, or leave it as is and I'll update it later, but thought I'd mention it as it might save you some time.

Reply:
I think I know how this CL needs to evolve. I guess I'll put this CL outside of this PR so I could focus on it separately. Sounds OK?..

Reply:
Yeh I think that makes sense, it should decouple easily enough.